### PR TITLE
Code quality: Compare null with 'is' instead of '=='

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
Comparing the null value using "is/is not" instead of "==/!=" is more efficient because it avoids comparing using the Equals method (which can be replaced by IEquatable<>, take resources, ...). This pr replaces all occurrences.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.App/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs [hunk 1]
```diff
@@ -88,9 +88,9 @@ public static class UniversalStorageEnumerator
...
                         {
                             var fileEntry = await AddFileAsync(item.AsBaseStorageFile(), currentStorageFolder, cancellationToken);
-                            if (fileEntry != null)
+                            if (fileEntry is not null)
                             {
-                                if (defaultIconPairs != null)
+                                if (defaultIconPairs is not null)
                                 {
                                     if (!string.IsNullOrEmpty(fileEntry.FileExtension))
                                     {
```
Comment (QuaintMako) [inline src/Files.App/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs:96]: Not the initial scope of the PR, but if you could take 5mn to fuse the nested `if `here, that'd be awesome!

**Review by QuaintMako — APPROVED** [review]
LGTM, all the changes have their correct equivalence.

**Comment by QuaintMako** [comment]
Related to #4180

**Review by yaichenbaum — COMMENTED** [review]
Looks good!

It might be interesting to run a benchmark and see the difference.

**Review by yaichenbaum — APPROVED** [review]

**Comment by cinqmilleans** [comment]
Merge fixed.

**Review by hez2010 — APPROVED** [review]
LGTM

**Pull request merged**